src/cli/doc_subpath.rs

```diff
@@ -0,0 +1,122 @@
+use crate::errors::*;
+use std::ffi::OsString;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+struct DocData<'a> {
+    topic: &'a str,
+    subtopic: &'a str,
+    root: &'a Path,
+}
+
+fn no_document(topic: &str) -> Result<OsString> {
+    Err(format!("No document for '{}'", topic).into())
+}
+
+fn index_html(doc: &DocData, wpath: &Path) -> Option<OsString> {
+    let indexhtml = wpath.join("index.html");
+    match &doc.root.join(&indexhtml).exists() {
+        true => Some(indexhtml.into_os_string()),
```

Comment:
Any reason you don't make this function return a PathBuf and save the conversions ?